### PR TITLE
Persist sorting order for Memory Profile table

### PR DESCRIPTION
**Pull request by bkonyi**
Also changes default sort column to total size.

Fixes https://github.com/flutter/devtools/issues/4300

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/allocation_profile/allocation_profile_table_view.dart [hunk 1]
```diff
@@ -234,10 +234,14 @@ class _AllocationProfileTable extends StatelessWidget {
     ),
   ];
 
+  static final _initialSortColumn = _FieldSizeColumn(
+    heap: _HeapGeneration.total,
+  );
+
   static final _columns = [
     _FieldClassName(),
```
Comment (polina-c) [inline packages/devtools_app/lib/src/screens/memory/panes/allocation_profile/allocation_profile_table_view.dart:242]: Add postfix 'Column' for consistency to this class?

File: packages/devtools_app/lib/src/screens/memory/panes/allocation_profile/allocation_profile_table_view.dart [hunk 2]
```diff
@@ -265,12 +269,28 @@ class _AllocationProfileTable extends StatelessWidget {
...
+  final AllocationProfileTableViewController controller;
+
+  @override
+  State<_AllocationProfileTable> createState() =>
+      _AllocationProfileTableState();
+}
+
+class _AllocationProfileTableState extends State<_AllocationProfileTable> {
+  late SortDirection sortDirection;
+  late ColumnData<ClassHeapStats> sortColumn;
```
Comment (polina-c) [inline packages/devtools_app/lib/src/screens/memory/panes/allocation_profile/allocation_profile_table_view.dart:281]: Should these two fields be private?

Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/memory/panes/allocation_profile/allocation_profile_table_view.dart]: They definitely can be, I'm just a little bit looser on naming for members of private classes I guess... :-) Done.

File: packages/devtools_app/lib/src/screens/memory/panes/allocation_profile/allocation_profile_table_view.dart [hunk 3]
```diff
@@ -265,12 +269,28 @@ class _AllocationProfileTable extends StatelessWidget {
...
+  final AllocationProfileTableViewController controller;
+
+  @override
+  State<_AllocationProfileTable> createState() =>
+      _AllocationProfileTableState();
+}
+
+class _AllocationProfileTableState extends State<_AllocationProfileTable> {
+  late SortDirection _sortDirection;
+  late ColumnData<ClassHeapStats> _sortColumn;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/allocation_profile/allocation_profile_table_view.dart:281]: can you link to a bug to store these values in a TableController? or if one does not exist, can you create one? I believe this is now the second case where we are storing these value for a Table in the widget that creates the Table

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**